File: sixthSeminar/src/main/java/org/sopt/sixthSeminar/service/ServiceMemberService.java
package org.sopt.sixthSeminar.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.sopt.sixthSeminar.config.JwtTokenProvider;
import org.sopt.sixthSeminar.config.authentication.UserAuthentication;
import org.sopt.sixthSeminar.domain.ServiceMember;
import org.sopt.sixthSeminar.dto.request.serviceMember.ServiceMemberRequest;
import org.sopt.sixthSeminar.dto.response.member.MemberSignInResponse;
import org.sopt.sixthSeminar.exception.ErrorMessage;
import org.sopt.sixthSeminar.exception.model.AuthException;
import org.sopt.sixthSeminar.exception.model.NotFoundException;
import org.sopt.sixthSeminar.repository.ServiceMemberRepository;
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;


@Slf4j
@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class ServiceMemberService {

    private final ServiceMemberRepository serviceMemberJpaRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtTokenProvider jwtTokenProvider;
    private final AuthenticationManagerBuilder authenticationManagerBuilder;

    @Transactional
    public String create(final ServiceMemberRequest request) {
        ServiceMember serviceMember = ServiceMember.builder()
                .nickanme(request.nickname())
                .password(passwordEncoder.encode(request.password()))
                .build();
        serviceMemberJpaRepository.save(serviceMember);
        return serviceMember.getId().toString();
    }

       public ServiceMember findById(final Long memberId) {
        return serviceMemberJpaRepository.findById(memberId)
                .orElseThrow(() -> new NotFoundException(ErrorMessage.MEMBER_NOT_FOUND_EXCEPTION));
    }

    public MemberSignInResponse signIn(final ServiceMemberRequest request) {
        ServiceMember serviceMember = serviceMemberJpaRepository.findByNickname(request.nickname())
                .orElseThrow(() -> new NotFoundException(ErrorMessage.MEMBER_NOT_FOUND_EXCEPTION));

        UserAuthentication userAuthentication = new UserAuthentication(serviceMember.getId(), null, null);

        if (!passwordEncoder.matches(request.password(), serviceMember.getPassword())) {
            throw new AuthException(ErrorMessage.PASSWORD_INCORRECT_EXCEPTION);
        }

        return MemberSignInResponse.of(jwtTokenProvider.generateToken(userAuthentication));
    }

}

File: sixthKotlin/src/main/kotlin/org/sopt/sixthKotlin/service/ServiceMemberService.kt
package org.sopt.sixthKotlin.service

import lombok.extern.slf4j.Slf4j
import org.sopt.sixthKotlin.config.JwtTokenProvider
import org.sopt.sixthKotlin.config.authentication.UserAuthentication
import org.sopt.sixthKotlin.controller.dto.request.ServiceMemberRequest
import org.sopt.sixthKotlin.controller.dto.response.MemberSignInResponse
import org.sopt.sixthKotlin.domain.ServiceMember
import org.sopt.sixthKotlin.exception.ErrorMessage
import org.sopt.sixthKotlin.exception.SoptException.AuthorizationException
import org.sopt.sixthKotlin.exception.SoptException.MemberNotFoundException
import org.sopt.sixthKotlin.repository.ServiceMemberJpaRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class ServiceMemberService(
    private val serviceMemberJpaRepository: ServiceMemberJpaRepository,
    private val passwordEncoder: PasswordEncoder,
    private val jwtTokenProvider: JwtTokenProvider
) {
    @Transactional
    fun create(request: ServiceMemberRequest): String {
        val serviceMember = ServiceMember(request.nickname, passwordEncoder.encode(request.password))
        serviceMemberJpaRepository.save(serviceMember)
        return serviceMember.id.toString()
    }

    fun findById(memberId: Long): ServiceMember {
        return serviceMemberJpaRepository.findByIdOrNull(memberId) ?: throw MemberNotFoundException(ErrorMessage.MEMBER_NOT_FOUND)
    }

    fun signIn(request: ServiceMemberRequest): MemberSignInResponse {
        val serviceMember: ServiceMember =
            serviceMemberJpaRepository.findByNickname(request.nickname) ?: throw MemberNotFoundException(ErrorMessage.MEMBER_NOT_FOUND)
        val userAuthentication = UserAuthentication(serviceMember.id, null, null)
        if (!passwordEncoder.matches(request.password, serviceMember.password)) {
            throw AuthorizationException(ErrorMessage.PASSWORD_INCORRECT)
        }
        return MemberSignInResponse(jwtTokenProvider.generateToken(userAuthentication))
    }
}